presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -112,43 +127,65 @@ static Slice compactSlice(Slice slice, int index, int length)
         return Slices.copyOf(slice, index, length);
     }
 
-    static boolean[] compactBooleanArray(boolean[] array, int index, int length)
+    /**
+     * Returns the array starting at specified index and extends for the specified length.
```

Comment:
```
Returns an array containing elements in the specified range of the specified array.
If the range matches the entire array, the input array will be returned.
Otherwise, a copy will be returned.
```

Copy this to every overload